dimagi/commcare-hq
# Add Enterprise CommCare Client Version Compliance Report

## Product Description


New Tile
<img width="1438" alt="image" src="https://github.com/user-attachments/assets/17bd41e1-2b6a-48e0-921a-e700993534a1">

Report
<img width="678" alt="image" src="https://github.com/user-attachments/assets/cad5eaf0-0829-4c13-8df8-db22366a6bc9">

API response
<img width="1436" alt="image" src="https://github.com/user-attachments/assets/cf81bcd3-9408-47a6-8462-754c0fb7c095">


## Technical Summary

Ticket: https://dimagi.atlassian.net/browse/SAAS-16236

I'm a bit worried about the performance. When I test on production, it can take more than 2 minutes to generate total number for account like CRS

## Feature Flag

Doing iterative release, so no FF. Once it is merged, we release it.

## Safety Assurance

### Safety story

Tested locally, on staging, and also on production.

### Automated test coverage


Add test for util function I wrote

### QA Plan


QA Ticket: https://dimagi.atlassian.net/browse/QA-7315

Verify the tile works as expected:

- An enterprise account without any mobile workers should see '--' shown up in tile.

Email the report:
- A mobile worker doesn't have any device yet, and haven't submitted any forms yet, will not be included in the report
- A mobile worker whose latest submission is using the latest commcare version, will **not be included** in the report. 

- A mobile worker whose latest submission is using an older commcare version, ([instruction to install an older version](https://dimagi.atlassian.net/wiki/spaces/commcarepublic/pages/2143946363/Offline+Install+of+CommCare+on+Android)), this mobile worker **will be included** in the report. Then use the same mobile worker, but submitted a form using the "Log in as xx mobile worker" feature, verify this mobile worker **is not** in the report. Because I assume in this case we're always using the latest commcare version. (Using the same mobile worker is to verify the submission through "Log in as xx mobile worker" feature will replace the previous submission when calculate version number)

- Can a mobile worker submitted form using script? If yes, please verify if the latest submission is by script, this mobile worker **is not** in the report. Because I assume in this case we're always using the latest commcare version.

### Rollback instructions




### Labels & Review
